iree/hal/cuda/graph_command_buffer.c

```diff
@@ -11,13 +11,17 @@
 #include <stdint.h>
 
 #include "iree/base/api.h"
+#include "iree/base/internal/inline_array.h"
 #include "iree/base/tracing.h"
 #include "iree/hal/cuda/cuda_buffer.h"
+#include "iree/hal/cuda/cuda_event.h"
 #include "iree/hal/cuda/dynamic_symbols.h"
 #include "iree/hal/cuda/executable_layout.h"
 #include "iree/hal/cuda/native_executable.h"
 #include "iree/hal/cuda/status_util.h"
 
+#define CUDA_MAX_LEAF_NODES 16
```

Comment:
```suggestion
#define IREE_HAL_CUDA_MAX_LEAF_NODES 16
```
(so we don't ever conflict with CUDA itself)

Reply:
we'll probably want a higher value for this - maybe 64 to start - some models like the nlp training model have 40+ concurrent dispatches:
![image](https://user-images.githubusercontent.com/75337/142776260-994d26b4-5479-4117-a589-6924b7e24c8c.png)